src/device-types/HALight.cpp

```diff
@@ -122,13 +139,49 @@ bool HALight::setRGBColor(const RGBColor& color, const bool force)
     return false;
 }
 
+void HALight::setEffects(const char* const effects[], const uint8_t size)
+{
+    if (!(_features & EffectsFeature) || !effects || size == 0 || _effects) { // effects can be set only once
+        return;
+    }
+
+    _effects = new HASerializerArray(size, false);
+
+    uint8_t effectLen = 0;
+    for (uint8_t i = 0; i < size; i++) {
+        effectLen = strlen(effects[i]);
+
+        char* const effect = new char[effectLen + 1]; // include null terminator
+        effect[effectLen] = '\0';
+        memcpy(effect, effects[i], effectLen);
+
+        _effects->add(effect);
+    }
+}
+
+bool HALight::setEffect(const uint8_t effect, const bool force)
+{
+    if (!force && effect == _currentEffect) {
+        return true;
+    }
```

Comment:
In the function docs it states "Returns `true` if MQTT message has been published successfully." which is not really true in this case. So I suggest improving the documentation.

Reply:
Well-spotted :wink: :heavy_check_mark: